### PR TITLE
Add Timeout to CompletableFuture in GeyserBungeePingPassthrough to Prevent Memory Leak

### PR DESCRIPTION
**Pull request by ByteExceptionM**
This PR addresses a memory leak issue in the `GeyserBungeePingPassthrough` class. The `getPingInformation` method has been updated to include a timeout in the `CompletableFuture`. This change prevents unprocessed packets from accumulating in the queue, which previously led to blocked Netty threads and unreleased `ByteBufs`, ultimately preventing players from joining. The problem appears to be a network issue, although it is more likely that the ProxyPingEvent is being blocked. This prevents Geyser from retrieving the MOTD and responding to appropriate requests. Bedrock needs the MOTD before a connection can be established. In addition, the network threads could be overloaded by this blocking, which leads to further connection issues.

### Changes Made:

**Added Timeout to `CompletableFuture`:**
   - Implemented a timeout to ensure that packets are processed within a reasonable timeframe.
   - This prevents the accumulation of unprocessed packets.

#### Unprocessed Packets in Queue
In the first image, you can see a high number of unprocessed UDP packets in the queue on port 19132. This is an indication of faulty packet processing. When data is sent to a device via a network protocol (such as TCP or UDP), this data is first written to a buffer before the receiving application reads it. The Recv-Q shows the size of this buffer in bytes. A high value means that a lot of data has been received that has not yet been processed by the application - Geyser in this case.

image

#### Packet Dump Analysis
This image shows a packet dump, which indicates that packets were sent to the machine and network interface, forwarded to the Docker network interface, but were not processed by Geyser.

image

#### Heap Dump Overview
The images provide an overview of the heap dump and memory usage of the proxy, highlighting the significant memory usage.

image
image
image

#### Problem Suspect 1
The next image and stack trace show Problem Suspect 1. The `epollEventLoopGroup-3-1` thread has a large memory footprint, retaining a substantial amount of memory.

image

```
epollEventLoopGroup-3-1
  at jdk.internal.misc.Unsafe.park(ZJ)V ((Native Method))
  at java.util.concurrent.locks.LockSupport.park(Ljava/lang/Object;)V ()
  at java.util.concurrent.CompletableFuture$Signaller.block()Z ()
  at java.util.concurrent.ForkJoinPool.unmanagedBlock(Ljava/util/concurrent/ForkJoinPool$ManagedBlocker;)V ()
  at java.util.concurrent.ForkJoinPool.managedBlock(Ljava/util/concurrent/ForkJoinPool$ManagedBlocker;)V ()
  at java.util.concurrent.CompletableFuture.waitingGet(Z)Ljava/lang/Object; ()
  at java.util.concurrent.CompletableFuture.join()Ljava/lang/Object; ()
  at org.geysermc.geyser.platform.bungeecord.GeyserBungeePingPassthrough.getPingInformation(Ljava/net/InetSocketAddress;)Lorg/geysermc/geyser/ping/GeyserPingInfo; (GeyserBungeePingPassthrough.java:62)
  at org.geysermc.geyser.network.netty.GeyserServer.onQuery(Lio/netty/channel/Channel;Ljava/net/InetSocketAddress;)Lorg/cloudburstmc/protocol/bedrock/BedrockPong; (GeyserServer.java:310)
  at org.geysermc.geyser.network.netty.handler.RakPingHandler.channelRead0(Lio/netty/channel/ChannelHandlerContext;Lorg/cloudburstmc/netty/channel/raknet/RakPing;)V (RakPingHandler.java:48)
  at org.geysermc.geyser.network.netty.handler.RakPingHandler.channelRead0(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (RakPingHandler.java:37)
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (SimpleChannelInboundHandler.java:99)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:444)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelHandlerContext; (AbstractChannelHandlerContext.java:412)
  at org.cloudburstmc.netty.handler.codec.raknet.server.RakServerOfflineHandler.onUnconnectedPing(Lio/netty/channel/ChannelHandlerContext;Lio/netty/channel/socket/DatagramPacket;Lio/netty/buffer/ByteBuf;J)V (RakServerOfflineHandler.java:134)
  at org.cloudburstmc.netty.handler.codec.raknet.server.RakServerOfflineHandler.channelRead0(Lio/netty/channel/ChannelHandlerContext;Lio/netty/channel/socket/DatagramPacket;)V (RakServerOfflineHandler.java:116)
  at org.cloudburstmc.netty.handler.codec.raknet.server.RakServerOfflineHandler.channelRead0(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (RakServerOfflineHandler.java:46)
  at org.cloudburstmc.netty.handler.codec.raknet.AdvancedChannelInboundHandler.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (AdvancedChannelInboundHandler.java:45)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:444)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelHandlerContext; (AbstractChannelHandlerContext.java:412)
  at org.cloudburstmc.netty.handler.codec.raknet.server.RakServerRateLimiter.channelRead0(Lio/netty/channel/ChannelHandlerContext;Lio/netty/channel/socket/DatagramPacket;)V (RakServerRateLimiter.java:148)
  at org.cloudburstmc.netty.handler.codec.raknet.server.RakServerRateLimiter.channelRead0(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (RakServerRateLimiter.java:35)
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (SimpleChannelInboundHandler.java:99)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:444)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelHandlerContext; (AbstractChannelHandlerContext.java:412)
  at org.geysermc.geyser.network.netty.handler.RakConnectionRequestHandler.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (RakConnectionRequestHandler.java:80)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:444)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelHandlerContext; (AbstractChannelHandlerContext.java:412)
  at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (DefaultChannelPipeline.java:1407)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:440)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.DefaultChannelPipeline.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelPipeline; (DefaultChannelPipeline.java:918)
  at org.cloudburstmc.netty.handler.codec.raknet.ProxyInboundRouter.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (ProxyInboundRouter.java:66)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:444)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelHandlerContext; (AbstractChannelHandlerContext.java:412)
  at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (DefaultChannelPipeline.java:1407)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:440)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.DefaultChannelPipeline.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelPipeline; (DefaultChannelPipeline.java:918)
  at io.netty.channel.epoll.EpollDatagramChannel.processPacket(Lio/netty/channel/ChannelPipeline;Lio/netty/channel/epoll/EpollRecvByteAllocatorHandle;ILio/netty/channel/socket/DatagramPacket;)V (EpollDatagramChannel.java:662)
  at io.netty.channel.epoll.EpollDatagramChannel.recvmsg(Lio/netty/channel/epoll/EpollRecvByteAllocatorHandle;Lio/netty/channel/epoll/NativeDatagramPacketArray;Lio/netty/buffer/ByteBuf;)Z (EpollDatagramChannel.java:697)
  at io.netty.channel.epoll.EpollDatagramChannel.access$300(Lio/netty/channel/epoll/EpollDatagramChannel;Lio/netty/channel/epoll/EpollRecvByteAllocatorHandle;Lio/netty/channel/epoll/NativeDatagramPacketArray;Lio/netty/buffer/ByteBuf;)Z (EpollDatagramChannel.java:56)
  at io.netty.channel.epoll.EpollDatagramChannel$EpollDatagramChannelUnsafe.epollInReady()V (EpollDatagramChannel.java:536)
  at io.netty.channel.epoll.EpollEventLoop.processReady(Lio/netty/channel/epoll/EpollEventArray;I)Z (EpollEventLoop.java:501)
  at io.netty.channel.epoll.EpollEventLoop.run()V (EpollEventLoop.java:399)
  at io.netty.util.concurrent.SingleThreadEventExecutor$4.run()V (SingleThreadEventExecutor.java:994)
  at io.netty.util.internal.ThreadExecutorMap$2.run()V (ThreadExecutorMap.java:74)
  at io.netty.util.concurrent.FastThreadLocalRunnable.run()V (FastThreadLocalRunnable.java:30)
  at java.lang.Thread.runWith(Ljava/lang/Object;Ljava/lang/Runnable;)V ()
  at java.lang.Thread.run()V ()
```

#### Problem Suspect 2
The next image shows Problem Suspect 2. The `epollEventLoopGroup-3-2` thread displays similar memory retention issues.

image

```
epollEventLoopGroup-3-2
  at jdk.internal.misc.Unsafe.park(ZJ)V ((Native Method))
  at java.util.concurrent.locks.LockSupport.park(Ljava/lang/Object;)V ()
  at java.util.concurrent.CompletableFuture$Signaller.block()Z ()
  at java.util.concurrent.ForkJoinPool.unmanagedBlock(Ljava/util/concurrent/ForkJoinPool$ManagedBlocker;)V ()
  at java.util.concurrent.ForkJoinPool.managedBlock(Ljava/util/concurrent/ForkJoinPool$ManagedBlocker;)V ()
  at java.util.concurrent.CompletableFuture.waitingGet(Z)Ljava/lang/Object; ()
  at java.util.concurrent.CompletableFuture.join()Ljava/lang/Object; ()
  at org.geysermc.geyser.platform.bungeecord.GeyserBungeePingPassthrough.getPingInformation(Ljava/net/InetSocketAddress;)Lorg/geysermc/geyser/ping/GeyserPingInfo; (GeyserBungeePingPassthrough.java:62)
  at org.geysermc.geyser.network.netty.GeyserServer.onQuery(Lio/netty/channel/Channel;Ljava/net/InetSocketAddress;)Lorg/cloudburstmc/protocol/bedrock/BedrockPong; (GeyserServer.java:310)
  at org.geysermc.geyser.network.netty.handler.RakPingHandler.channelRead0(Lio/netty/channel/ChannelHandlerContext;Lorg/cloudburstmc/netty/channel/raknet/RakPing;)V (RakPingHandler.java:48)
  at org.geysermc.geyser.network.netty.handler.RakPingHandler.channelRead0(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (RakPingHandler.java:37)
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (SimpleChannelInboundHandler.java:99)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:444)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelHandlerContext; (AbstractChannelHandlerContext.java:412)
  at org.cloudburstmc.netty.handler.codec.raknet.server.RakServerOfflineHandler.onUnconnectedPing(Lio/netty/channel/ChannelHandlerContext;Lio/netty/channel/socket/DatagramPacket;Lio/netty/buffer/ByteBuf;J)V (RakServerOfflineHandler.java:134)
  at org.cloudburstmc.netty.handler.codec.raknet.server.RakServerOfflineHandler.channelRead0(Lio/netty/channel/ChannelHandlerContext;Lio/netty/channel/socket/DatagramPacket;)V (RakServerOfflineHandler.java:116)
  at org.cloudburstmc.netty.handler.codec.raknet.server.RakServerOfflineHandler.channelRead0(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (RakServerOfflineHandler.java:46)
  at org.cloudburstmc.netty.handler.codec.raknet.AdvancedChannelInboundHandler.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (AdvancedChannelInboundHandler.java:45)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:444)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelHandlerContext; (AbstractChannelHandlerContext.java:412)
  at org.cloudburstmc.netty.handler.codec.raknet.server.RakServerRateLimiter.channelRead0(Lio/netty/channel/ChannelHandlerContext;Lio/netty/channel/socket/DatagramPacket;)V (RakServerRateLimiter.java:148)
  at org.cloudburstmc.netty.handler.codec.raknet.server.RakServerRateLimiter.channelRead0(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (RakServerRateLimiter.java:35)
  at io.netty.channel.SimpleChannelInboundHandler.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (SimpleChannelInboundHandler.java:99)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:444)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelHandlerContext; (AbstractChannelHandlerContext.java:412)
  at org.geysermc.geyser.network.netty.handler.RakConnectionRequestHandler.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (RakConnectionRequestHandler.java:80)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:444)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelHandlerContext; (AbstractChannelHandlerContext.java:412)
  at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (DefaultChannelPipeline.java:1407)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:440)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.DefaultChannelPipeline.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelPipeline; (DefaultChannelPipeline.java:918)
  at org.cloudburstmc.netty.handler.codec.raknet.ProxyInboundRouter.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (ProxyInboundRouter.java:66)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:444)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelHandlerContext; (AbstractChannelHandlerContext.java:412)
  at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(Lio/netty/channel/ChannelHandlerContext;Ljava/lang/Object;)V (DefaultChannelPipeline.java:1407)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:440)
  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(Lio/netty/channel/AbstractChannelHandlerContext;Ljava/lang/Object;)V (AbstractChannelHandlerContext.java:420)
  at io.netty.channel.DefaultChannelPipeline.fireChannelRead(Ljava/lang/Object;)Lio/netty/channel/ChannelPipeline; (DefaultChannelPipeline.java:918)
  at io.netty.channel.epoll.EpollDatagramChannel.processPacket(Lio/netty/channel/ChannelPipeline;Lio/netty/channel/epoll/EpollRecvByteAllocatorHandle;ILio/netty/channel/socket/DatagramPacket;)V (EpollDatagramChannel.java:662)
  at io.netty.channel.epoll.EpollDatagramChannel.recvmsg(Lio/netty/channel/epoll/EpollRecvByteAllocatorHandle;Lio/netty/channel/epoll/NativeDatagramPacketArray;Lio/netty/buffer/ByteBuf;)Z (EpollDatagramChannel.java:697)
  at io.netty.channel.epoll.EpollDatagramChannel.access$300(Lio/netty/channel/epoll/EpollDatagramChannel;Lio/netty/channel/epoll/EpollRecvByteAllocatorHandle;Lio/netty/channel/epoll/NativeDatagramPacketArray;Lio/netty/buffer/ByteBuf;)Z (EpollDatagramChannel.java:56)
  at io.netty.channel.epoll.EpollDatagramChannel$EpollDatagramChannelUnsafe.epollInReady()V (EpollDatagramChannel.java:536)
  at io.netty.channel.epoll.EpollEventLoop.processReady(Lio/netty/channel/epoll/EpollEventArray;I)Z (EpollEventLoop.java:501)
  at io.netty.channel.epoll.EpollEventLoop.run()V (EpollEventLoop.java:399)
  at io.netty.util.concurrent.SingleThreadEventExecutor$4.run()V (SingleThreadEventExecutor.java:994)
  at io.netty.util.internal.ThreadExecutorMap$2.run()V (ThreadExecutorMap.java:74)
  at io.netty.util.concurrent.FastThreadLocalRunnable.run()V (FastThreadLocalRunnable.java:30)
  at java.lang.Thread.runWith(Ljava/lang/Object;Ljava/lang/Runnable;)V ()
  at java.lang.Thread.run()V ()
```

### Additional Information:
- This problem also caused players to be unable to join because of blocked Netty threads. Player connections were not being correctly terminated, leading to further complications in the server operation.
- Since implementing the timeout, no further issues have been observed.
- Extensive testing has been conducted to ensure that the timeout effectively mitigates the memory leak and unprocessed packet issues. However, I will continue to monitor the problem.

Please review the changes and let me know if any further adjustments are needed.

### CODE REVIEW

File: bootstrap/bungeecord/src/main/java/org/geysermc/geyser/platform/bungeecord/GeyserBungeePingPassthrough.java [hunk 1]
```diff
@@ -59,7 +61,17 @@ public GeyserPingInfo getPingInformation(InetSocketAddress inetSocketAddress) {
...
             }
         }));
-        ProxyPingEvent event = future.join();
+
+        ProxyPingEvent event;
+
+        try {
+            event = future.get(100, TimeUnit.MILLISECONDS);
+        } catch (Throwable cause) {
+            proxyServer.getLogger().log(Level.SEVERE, "Failed to get ping information for " + inetSocketAddress, cause);
```
Comment (onebeastchris) [inline bootstrap/bungeecord/src/main/java/org/geysermc/geyser/platform/bungeecord/GeyserBungeePingPassthrough.java:70]: Please use the Geyser logger here - you can access it via `GeyserImpl.getInstance().getLogger()`.

**Review by onebeastchris — CHANGES_REQUESTED** [review]

File: bootstrap/bungeecord/src/main/java/org/geysermc/geyser/platform/bungeecord/GeyserBungeePingPassthrough.java [hunk 2]
```diff
@@ -59,7 +61,17 @@ public GeyserPingInfo getPingInformation(InetSocketAddress inetSocketAddress) {
...
         }));
-        ProxyPingEvent event = future.join();
+
+        ProxyPingEvent event;
+
+        try {
+            event = future.get(100, TimeUnit.MILLISECONDS);
+        } catch (Throwable cause) {
+            GeyserImpl.getInstance().getLogger().error("Failed to get ping information for " + inetSocketAddress, cause);
+
```
Comment (Konicai) [inline bootstrap/bungeecord/src/main/java/org/geysermc/geyser/platform/bungeecord/GeyserBungeePingPassthrough.java:71]: would probably be wise to do something similar to this: https://github.com/GeyserMC/Geyser/blob/9cdda70/core/src/main/java/org/geysermc/geyser/session/GeyserSession.java#L1092

File: core/src/main/java/org/geysermc/geyser/ping/IGeyserPingPassthrough.java [hunk 3]
```diff
@@ -35,10 +35,10 @@
 public interface IGeyserPingPassthrough {
 
     /**
-     * Get the MOTD of the server displayed on the multiplayer screen
+     * Get the ping information, including the MOTD and player count, from the server
```
Comment (Konicai) [inline core/src/main/java/org/geysermc/geyser/ping/IGeyserPingPassthrough.java:38]:
```suggestion
     * Gets the ping information, including the MOTD and player count, from the server
```

**Review by Tjorven-Liebe — APPROVED** [review]
This seems like a legitimate way to fix the memory leak. Approved.

**Review by JonasDevDE — APPROVED** [review]
nice fix.

File: bootstrap/bungeecord/src/main/java/org/geysermc/geyser/platform/bungeecord/GeyserBungeePingPassthrough.java [hunk 4]
```diff
@@ -59,7 +61,19 @@ public GeyserPingInfo getPingInformation(InetSocketAddress inetSocketAddress) {
...
+        ProxyPingEvent event;
+
+        try {
+            event = future.get(100, TimeUnit.MILLISECONDS);
+        } catch (Throwable cause) {
+            String address = GeyserImpl.getInstance().getConfig().isLogPlayerIpAddresses() ? inetSocketAddress.toString() : "<IP address withheld>";
+
+            GeyserImpl.getInstance().getLogger().error("Failed to get ping information for " + address, cause);
+
```
Comment (onebeastchris) [inline bootstrap/bungeecord/src/main/java/org/geysermc/geyser/platform/bungeecord/GeyserBungeePingPassthrough.java:73]:
```suggestion
```

File: bootstrap/bungeecord/src/main/java/org/geysermc/geyser/platform/bungeecord/GeyserBungeePingPassthrough.java [hunk 5]
```diff
@@ -59,7 +61,19 @@ public GeyserPingInfo getPingInformation(InetSocketAddress inetSocketAddress) {
...
         }));
-        ProxyPingEvent event = future.join();
+
+        ProxyPingEvent event;
+
+        try {
+            event = future.get(100, TimeUnit.MILLISECONDS);
+        } catch (Throwable cause) {
+            String address = GeyserImpl.getInstance().getConfig().isLogPlayerIpAddresses() ? inetSocketAddress.toString() : "<IP address withheld>";
+
```
Comment (onebeastchris) [inline bootstrap/bungeecord/src/main/java/org/geysermc/geyser/platform/bungeecord/GeyserBungeePingPassthrough.java:71]:
```suggestion
```

**Review by onebeastchris — APPROVED** [review]
two small nitpicks, lgtm otherwise

**Pull request merged**